src/bin/main.rs

```diff
@@ -342,6 +346,8 @@ fn main() {
         Ok(summary) => {
             if summary.has_operational_errors() {
                 1
+            } else if summary.has_diff && summary.write_mode == Some(WriteMode::Check) {
+                1
```

Comment:
It will probably be easier to change the other exit codes now, rather than later.

Reply:
Alrighty! I'll implement the stuff suggested in this comment: https://github.com/rust-lang-nursery/rustfmt/issues/1977#issuecomment-372167773